AirCasting/Settings/BackendURLValidator.swift

```diff
@@ -0,0 +1,70 @@
+//
+//  BackendURLValidator.swift
+//  URLValidation
+//
+//  Created by Pawel Gil on 25/06/2021.
+//
+
+import Foundation
+
+class BackendURLBuilder {
+    enum ValidationError: Error {
+        case invalidURL
+        case invalidPort
+        case noURL
+    }
+    
+    func createURL(url: String, port: String) throws -> URL? {
+        if url.count == 0 && port.count == 0 { return nil }
+        if url.count == 0 && port.count > 0 { throw ValidationError.noURL }
+        let url = appenURLWithSchemeIfNotPresent(url: url)
+        var components = try getComponents(url: url)
+        try validateHost(for: components)
+        components.port = try getPortNumber(from: port)
+        return components.url!
+    }
+    
+    private func getComponents(url: String) throws -> URLComponents {
+        guard let components = URLComponents(string: url) else {
+            throw ValidationError.invalidURL
+        }
+        return components
+    }
+    
+    private func validateHost(for components: URLComponents) throws {
+        let host = try getHost(from: components)
+        if host.contains(".") {
```

Comment:
great idea with the `separatedBy` usage! It's a bit convoluted tho, will this work ok perhaps?:

```Swift
let hostComponents = getHost(from: components).components(separatedBy: ".")
guard components.count > 1, components.allSatisfy({ $0.count > 0 }) else {
    throw ValidationError.invalidURL
    return
}
```

In essence this code:
1. Checks if we have at least 2 components separated by dots in a host string
2. Checks if all of those components are at last one character

Basically this requires each dot to have something before and something after. 
Note this won't protect against all unwanted URL's (for this we'd need a complex regex), but this is not our goal here I think. Our goal is to not enable the button until user types in some text, a dot, and some more text. So I think this will work fine:

`test.c` -> button enabled
`test.` -> button disabled
`.test` -> button disabled
etc.